src/main/java/io/jenkins/plugins/appcenter/api/AppCenterServiceFactory.java

```diff
@@ -88,6 +89,56 @@ public UploadService createUploadService(@Nonnull final String uploadUrl) {
         return retrofit.create(UploadService.class);
     }
 
+    public BlobClient createBlobUploadService(@Nonnull final String uploadUrl) {
+        //Workaround for bug in Azure Blob Storage, as AppCenter returns the upload URL with a port attached
+        //See https://github.com/Azure/azure-sdk-for-java/issues/15827
+        final HttpUrl httpUploadUrl = HttpUrl.get(uploadUrl);
+        final String file;
+        if (httpUploadUrl.encodedQuery() != null) {
```

Comment:
This should be the same as the workaround you had but using `HttpUrl` instead of `URL` from Java. This allows us to use the same semantics as in the rest of this class and also omit the try-catch block.